### PR TITLE
Fix missing `base_score` in xgboost regression

### LINKED ISSUE #3579: BUG: `base_score` attribute of the `XGBTreeModelLoader` is broken for all exponential losses (e.g. tweedie, poisson)
### Issue Description

When trying to run a `TreeExplainer` for a XGBoost model setup with the `tweedie` loss, I get an error stating that some internal `self.base_score` of the `TreeExplainer` object cannot be found.

### Minimal Reproducible Example

```python
from xgboost import XGBRegressor
import shap
import numpy as np

X, y = np.random.randn(100, 5), np.random.exponential(size=100)
model = XGBRegressor(
    objective="reg:tweedie",
)
model.fit(X, y)
explainer = shap.TreeExplainer(model)
```

### Traceback

```shell
File .../xgboost_error.py:7
      3 model = XGBRegressor(
      4     objective="reg:tweedie"
      5 )
      6 model.fit(X, y)
----> 7 explainer = shap.TreeExplainer(model)

File .../site-packages/shap/explainers/_tree.py:195, in TreeExplainer.__init__(self, model, data, model_output, feature_perturbation, feature_names, approximate, link, linearize_link)
    193 self.feature_perturbation = feature_perturbation
    194 self.expected_value = None
--> 195 self.model = TreeEnsemble(model, self.data, self.data_missing, model_output)
    196 self.model_output = model_output
    197 #self.model_output = self.model.model_output # this allows the TreeEnsemble to translate model outputs types by how it loads the model

File .../site-packages/shap/explainers/_tree.py:1090, in TreeEnsemble.__init__(self, model, data, data_missing, model_output)
   1088 elif safe_isinstance(model, "xgboost.sklearn.XGBRegressor"):
   1089     self.original_model = model.get_booster()
-> 1090     self._set_xgboost_model_attributes(
   1091         data,
   1092         data_missing,
   1093         objective_name_map,
   1094         tree_output_name_map,
   1095     )
   1096     # Some properties of the sklearn API are passed to a DMatrix object in
   1097     # xgboost We need to make sure we do the same here - GH #3313
   1098     self._xgb_dmatrix_props = get_xgboost_dmatrix_properties(model)

File .../site-packages/shap/explainers/_tree.py:1315, in TreeEnsemble._set_xgboost_model_attributes(self, data, data_missing, objective_name_map, tree_output_name_map)
   1308 def _set_xgboost_model_attributes(
   1309     self,
   1310     data,
   1311     data_missing, objective_name_map,
   1312     tree_output_name_map,
   1313 ):
   1314     self.model_type = "xgboost"
-> 1315     loader = XGBTreeModelLoader(self.original_model)
   1317     self.trees = loader.get_trees(data=data, data_missing=data_missing)
   1318     self.base_offset = loader.base_score

File .../site-packages/shap/explainers/_tree.py:1873, in XGBTreeModelLoader.__init__(self, xgb_model)
   1864     self.base_score = scipy.special.logit(base_score)
   1865 elif self.name_obj in (
   1866     "reg:gamma",
   1867     "reg:tweedie",
   (...)
   1871 ):
   1872     # exp family
-> 1873     self.base_score = np.log(self.base_score)
   1874 else:
   1875     self.base_score = base_score
```

### Expected Behavior

The `base_score` attribute should be initialised properly with no error.

The line 1873 should be `self.base_score = np.log(base_score)` without referring to `self`.

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [X] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [X] I'd be interested in making a PR to fix this bug

### Installed Versions

shap==0.45.0

### PR DESCRIPTION
**Pull request by CloseChoice**
## Overview

Closes #3579  

Description of the changes proposed in this pull request:
 - initialize `base_score` for xgboost
 - add test

## Checklist

- [x] All [pre-commit checks](https://pre-commit.com/#install) pass.
- [x] Unit tests added (if fixing a bug or adding a new feature)

### CODE REVIEW

**Comment by CloseChoice** [comment]
@connortann would be nice if we could get a new hotfix release when this is merged, since it seems to be needed by the community

**Comment by CloseChoice** [comment]
hmm, seems like our macos pipeline is failing due to xgboost errors. Haven't we had this before?

**Review by connortann — APPROVED** [review]
LGTM!
